src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -226,13 +228,30 @@ void GcsNodeManager::HandleGetAllNodeInfo(rpc::GetAllNodeInfoRequest request,
     filter_state = request.filters().state();
   }
   std::string filter_node_name = request.filters().node_name();
-  auto filter_fn = [&filter_node_id, &filter_node_name](const rpc::GcsNodeInfo &node) {
+  std::string filter_virtual_cluster_id = request.filters().virtual_cluster_id();
+  RAY_LOG(DEBUG) << "GetAllNodeInfo request, limit = " << limit
+                 << ", filter_node_id = " << filter_node_id
+                 << ", filter_node_name = " << filter_node_name
+                 << ", filter_virtual_cluster_id = " << filter_virtual_cluster_id;
+  auto filter_fn = [this, &filter_node_id, &filter_node_name, &filter_virtual_cluster_id](const rpc::GcsNodeInfo &node) {
     if (!filter_node_id.IsNil() && filter_node_id != NodeID::FromBinary(node.node_id())) {
       return false;
     }
     if (!filter_node_name.empty() && filter_node_name != node.node_name()) {
       return false;
     }
+    if (filter_virtual_cluster_id.empty()) {
+      return true;
+    }
+    auto virtual_cluster = gcs_virtual_cluster_manager_.GetVirtualCluster(filter_virtual_cluster_id);
+    if (virtual_cluster != nullptr) {
```

Comment:
if (virtual_cluster == nullptr) {
  return false;
}

bool contains = virtual_cluster->ContainsNodeInstance(NodeID::FromBinary(node.node_id()).Hex());
      RAY_LOG(DEBUG) << "Filtering node " << node.node_id() << " in virtual cluster "
                << filter_virtual_cluster_id << " contains: " << contains;
return contains;